src/Emulator/Peripherals/Peripherals/Miscellaneous/OpenTitan_LifeCycleController.cs

```diff
@@ -87,9 +87,12 @@ public string DeviceId
                 // This array was reversed to comply with the OpenTitan tests suite - the spec does not specify this
                 deviceId = Misc.HexStringToByteArray(value).Reverse().ToArray();
                 var otpDeviceId = otpController.GetOtpItem(OpenTitan_OneTimeProgrammableMemoryController.OtpItem.DeviceId);
-                if(deviceId != otpDeviceId)
+                // Only print the warning message if otpDeviceId is set.
+                if(deviceId != otpDeviceId && !otpDeviceId.All(o => o == 0))
```

Comment:
Just a small suggestion:
```suggestion
                if(deviceId != otpDeviceId && otpDeviceId.Any(o => o != 0))
```

Reply:
Actually, the check should be `All`. `otpDeviceId` is a byte array, and if the ID is not set, ti is a zero array, so all the elements are zero, instead of any one of them being 0.

Reply:
I also reverted the condition inside lambda and removed the negation before the whole `otpDeviceId.Any` so I think it's equivalent, but your solution works fine as well.